crates/ruff_linter/src/rules/refurb/rules/helpers.rs

```diff
@@ -0,0 +1,40 @@
+use std::borrow::Cow;
+
+use ruff_python_ast::{self as ast, parenthesize::parenthesized_range};
+
+use crate::checkers::ast::Checker;
+
+/// A helper function that extracts the `iter` from a [`ast::StmtFor`] node and,
+/// if the `iter` is an unparenthesized tuple, adds parentheses:
+///
+/// - `for x in z: ...`       ->  `"x"`
+/// - `for (x, y) in z: ...`  ->  `"(x, y)"`
+/// - `for [x, y] in z: ...`  ->  `"[x, y]"`
+/// - `for x, y in z: ...`    ->  `"(x, y)"`      # <-- Parentheses added only for this example
+pub(super) fn parenthesize_loop_iter_if_necessary<'a>(
+    for_stmt: &'a ast::StmtFor,
+    checker: &'a Checker,
+) -> Cow<'a, str> {
```

Comment:
Using a `Cow` here is nice but it feels a bit overkill, considering that this is a fix.